src/effects/VST/VSTEffect.cpp

```diff
@@ -1911,22 +1911,35 @@ std::vector<int> VSTEffect::GetEffectIDs()
    return effectIDs;
 }
 
+
+VstPatchChunkInfo VSTEffectWrapper::GetChunkInfo() const
+{
+   VstPatchChunkInfo info = { 1, mAEffect->uniqueID, mAEffect->version, mAEffect->numParams, "" };
+   return info;
+}
+
+bool VSTEffectWrapper::IsCompatible(const VstPatchChunkInfo& info) const
+{
+   return  (info.pluginUniqueID == mAEffect->uniqueID) &&
+           (info.pluginVersion  == mAEffect->version) &&
+           (info.numElements    == mAEffect->numParams);
+}
+
 bool VSTEffect::LoadParameters(
    const RegistryPath & group, EffectSettings &settings)
 {
    wxString value;
 
-   VstPatchChunkInfo info = {1, mAEffect->uniqueID, mAEffect->version, mAEffect->numParams, ""};
+   auto info = GetChunkInfo();
+
```

Comment:
The commit that rewrites this function can also make it and `StoreCommandParameters` const, and that will allow you to remove `DoLoadUserPreset` and a `const_cast`

Reply:
ok, done